common/src/main/java/com/viaversion/viaversion/protocols/protocol1_16to1_15_2/packets/EntityPackets.java

```diff
@@ -42,25 +43,35 @@
 public class EntityPackets {
 
     private static final PacketHandler DIMENSION_HANDLER = wrapper -> {
+        WorldIdentifiers map = Via.getConfig().get1_16WorldNamesMap();
+        WorldIdentifiers userMap = wrapper.user().get(WorldIdentifiers.class);
+        if (userMap!=null){
+            map = userMap;
+        }
```

Comment:
Is there actually a need for per player world identifiers for other plugins to set?

Reply:
I think so - that was the primary motivation for this second set of changes. For example, in Shotbow, one of their game modes has the 'lobby' and the actual map - and you get sent back to the lobby when you die.
As far as I'm aware, these are on the same server with the same ViaVersion version, but not all players are in the same one.